### DISCUSSION TITLE
Memory Fragmentation

### DISCUSSION THREAD

**Discussion by M-Hammod** [protobuf-c/protobuf-c#640] [Q&A] [answered]
I'm using protobuf-c library in an embedded device that could run for long time, although the device has enough RAM I'm worried about memory fragmentation, I wonder if there is anything I can do to configure the library cmessage__unpack to always allocate big chunk of continuous memory even if the received message is very small.

**Comment by edmonds** [comment, marked as answer]
If you want to replace the allocator used to satisfy memory allocations during unpacking, use the `ProtobufCAllocator *` parameter to the unpack function.

If you are looking for a protobuf decoder that doesn't perform dynamic memory allocations, that's not `protobuf-c` but [nanopb](https://koti.kapsi.fi/jpa/nanopb/).
